### DISCUSSION TITLE
Sounds from Train - City Hubs

### DISCUSSION THREAD

**Discussion by TekyTeky** [Novakasa/brickrail#168] [Ideas]
I wasn't aware if this is possible but would be great to have. 
I have sounds in BAP, when a train goes over a coloured tile it plays a sound on the PC. Eg approaching a station it maybe a horn or railway crossing maybe bells. 
I just saw this and wondered:

https://www.eurobricks.com/forum/index.php?/forums/topic/186132-pybricks-qa/&do=findComment&comment=3652214

**Comment by Novakasa** [comment]
The City hub does not have a speaker, only the spike prime / robot inventor hub has a speaker as far as I know. Since probably no one uses these hubs in trains, I don't think I will implement it, also it would probably complicate the firmware/program download step since you need to get the sound files on there.

Adding sound playback by PC would be of course possible. Crossing bell could automatically play when a crossing closes, and train station sounds could also be context sensitive since the software knows if a train will brake or stop. Sounds triggered just via markers is also possible.

**Reply by Novakasa** [reply]
Theoretically, there is probably a way to implement multi-channel output in case people want to setup surround speakers or a custom solution to place speakers at different positions. I don't plan to do that with my setup, so would be hard to test, but it is probably doable in theory.

**Comment by Novakasa** [comment]
tracked as a feature request in https://github.com/Novakasa/brickrail/issues/169
